### PR TITLE
[@types/expo] Fix Segment analytics type definitions

### PR DESCRIPTION
**Pull request by romain-faust**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/expo/expo/blob/master/packages/expo-analytics-segment/src/Segment.ts
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/expo/index.d.ts [hunk 1]
```diff
@@ -2508,21 +2509,30 @@ export namespace SecureStore {
...
     function identify(userId: string): void;
     function identifyWithTraits(userId: string, traits: object): void;
-    function track(event: string): void;
-    function alias(newId: string, options?: { [key: string]: any }): Promise<boolean>;
+    function group(groupId: string): void;
+    function groupWithTraits(groupId: string, traits: object): void;
     function reset(): void;
-    function trackWithProperties(event: string, properties: object): void;
+
+    function alias(newId: string, options?: object): Promise<boolean>;
```
Comment (moshfeu) [inline types/expo/index.d.ts]: Why object?
https://github.com/expo/expo/blob/58d73bf85785b74f5659edb1c743e79d72a4b22c/packages/expo-analytics-segment/src/Segment.ts#L49

Comment (romain-faust) [reply types/expo/index.d.ts]: To harmonize the signature :
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/expo/index.d.ts#L2537
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/expo/index.d.ts#L2541
- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/expo/index.d.ts#L2543

Comment (moshfeu) [reply types/expo/index.d.ts]: So maybe these should be change too.

https://github.com/expo/expo/blob/58d73bf85785b74f5659edb1c743e79d72a4b22c/packages/expo-analytics-segment/src/Segment.ts#L70

Comment (romain-faust) [reply types/expo/index.d.ts]: Yes :)

Comment (moshfeu) [reply types/expo/index.d.ts]: Want to do this :)?

Comment (romain-faust) [reply types/expo/index.d.ts]: Yes why not :)

Comment (romain-faust) [reply types/expo/index.d.ts]: https://github.com/expo/expo/pull/4116

Comment (moshfeu) [reply types/expo/index.d.ts]: 👍

**Review by moshfeu — APPROVED** [review]
LGTM

**Pull request merged**